pkg/ddc/jindo/load_data.go

```diff
@@ -0,0 +1,181 @@
+	"os"
+	"strings"
+)
+
+// CreateDataLoadJob creates the job to load data
+func (e *JindoEngine) CreateDataLoadJob(ctx cruntime.ReconcileRequestContext, targetDataload datav1alpha1.DataLoad) (err error) {
+	log := ctx.Log.WithName("createDataLoadJob")
+
+	// 1. Check if the helm release already exists
+	releaseName := utils.GetDataLoadReleaseName(targetDataload.Name)
+	jobName := utils.GetDataLoadJobName(releaseName)
+	var existed bool
+	existed, err = helm.CheckRelease(releaseName, targetDataload.Namespace)
+	if err != nil {
+		log.Error(err, "failed to check if release exists", "releaseName", releaseName, "namespace", targetDataload.Namespace)
+		return err
+	}
+
+	// 2. install the helm chart if not exists
+	if !existed {
+		log.Info("DataLoad job helm chart not installed yet, will install")
+		valueFileName, err := e.generateDataLoadValueFile(ctx, targetDataload)
+		if err != nil {
+			log.Error(err, "failed to generate dataload chart's value file")
+			return err
+		}
+		chartName := utils.GetChartsDirectory() + "/" + cdataload.DATALOAD_CHART + "/" + common.JINDO_RUNTIME
+		err = helm.InstallRelease(releaseName, targetDataload.Namespace, valueFileName, chartName)
+		if err != nil {
+			log.Error(err, "failed to install dataload chart")
+			return err
+		}
+		log.Info("DataLoad job helm chart successfully installed", "namespace", targetDataload.Namespace, "releaseName", releaseName)
+		ctx.Recorder.Eventf(&targetDataload, v1.EventTypeNormal, common.DataLoadJobStarted, "The DataLoad job %s started", jobName)
+	}
+	return err
+}
+
+// generateDataLoadValueFile builds a DataLoadValue by extracted specifications from the given DataLoad, and
+// marshals the DataLoadValue to a temporary yaml file where stores values that'll be used by fluid dataloader helm chart
+func (e *JindoEngine) generateDataLoadValueFile(r cruntime.ReconcileRequestContext, dataload datav1alpha1.DataLoad) (valueFileName string, err error) {
+	targetDataset, err := utils.GetDataset(r.Client, dataload.Spec.Dataset.Name, dataload.Spec.Dataset.Namespace)
+	if err != nil {
+		return "", err
+	}
+
+	_, boundedRuntime := utils.GetRuntimeByCategory(targetDataset.Status.Runtimes, common.AccelerateCategory)
+
+	imageName, imageTag := docker.GetWorkerImage(r.Client, dataload.Spec.Dataset.Name, boundedRuntime.Type, dataload.Spec.Dataset.Namespace)
+
+	if len(imageName) == 0 {
+		defaultImageInfo := strings.Split(common.DEFAULT_JINDO_RUNTIME_IMAGE, ":")
+		if len(defaultImageInfo) < 1 {
+			panic("invalid default dataload image!")
+		} else {
+			imageName = defaultImageInfo[0]
+		}
+	}
+
+	if len(imageTag) == 0 {
+		defaultImageInfo := strings.Split(common.DEFAULT_JINDO_RUNTIME_IMAGE, ":")
+		if len(defaultImageInfo) < 2 {
+			panic("invalid default dataload image!")
+		} else {
+			imageTag = defaultImageInfo[1]
+		}
+	}
+
+	image := fmt.Sprintf("%s:%s", imageName, imageTag)
+
+	runtime, err := utils.GetJindoRuntime(r.Client, boundedRuntime.Name, boundedRuntime.Namespace)
```

Comment:
Better to reuse `r.Runtime` with a type conversion here to get the JindoRuntime here

Reply:
thx and I will have a try